frontend/packages/schema-model/src/lib/mutations/ui-schema-reducers.test.ts

```diff
@@ -367,6 +367,32 @@ describe('ui-schema-reducers', () => {
       expect(updatedNode.fieldType).toEqual(stringNodeMock.fieldType);
       expect(updatedNode.isArray).toBe(true);
     });
+
+    it("Updates the childrens' pointers", () => {
+      const { pointer } = parentNodeMock;
+      result = toggleArrayField(uiSchemaMock, pointer);
+      const updatedNode = getNodeByPointer(result, pointer);
+      expect(updatedNode.children.length).toEqual(parentNodeMock.children.length);
+      updatedNode.children.forEach((childPointer) => {
+        expect(childPointer.startsWith(`${pointer}/${Keyword.Items}`)).toBe(true);
+      });
+      getChildNodesByPointer(result, pointer).forEach((childNode) => {
+        expect(childNode.pointer.startsWith(`${pointer}/${Keyword.Items}`)).toBe(true);
+      });
+    });
+
+    it("should  not update the children's pointers with /Items/ when isArray OFF", () => {
+      const { pointer } = parentNodeMock;
+      result = toggleArrayField(uiSchemaMock, pointer);
+      const updatedNode = getNodeByPointer(result, pointer);
+      expect(updatedNode.children.length).toEqual(parentNodeMock.children.length);
+      updatedNode.children.forEach((childPointer) => {
+        expect(childPointer.startsWith(`${pointer}`)).toBe(true);
```

Comment:
This will be true in both cases (toggling on and off), so this test does not tell us much. You should check that the pointer does NOT start with `${pointer}/${Keyword.Items}`